src/wazuh_modules/sca/sca_impl/src/sca_impl.cpp

```diff
@@ -1059,4 +1213,145 @@ bool SecurityConfigurationAssessment::integrityIntervalElapsed(int64_t currentTi
     return intervalElapsed;
 }
 
+bool SecurityConfigurationAssessment::hasDataInDatabase()
+{
+    if (!m_dBSync)
+    {
+        LoggingHelper::getInstance().log(LOG_ERROR, "DBSync is null, cannot check database contents");
+        return false;
+    }
+
+    try
+    {
+        int policyCount = 0;
+        int checkCount = 0;
+
+        // Count policies
+        auto policyQuery = SelectQuery::builder()
+                           .table("sca_policy")
+                           .columnList({"COUNT(*) AS count"})
+                           .build();
+
+        const auto policyCallback = [&policyCount](ReturnTypeCallback returnTypeCallback, const nlohmann::json & resultData)
+        {
+            if (returnTypeCallback == SELECTED && resultData.contains("count"))
+            {
+                if (resultData["count"].is_number())
+                {
+                    policyCount = resultData["count"].get<int>();
+                }
+            }
+        };
+
+        m_dBSync->selectRows(policyQuery.query(), policyCallback);
+
+        // Count checks
+        auto checkQuery = SelectQuery::builder()
+                          .table("sca_check")
+                          .columnList({"COUNT(*) AS count"})
+                          .build();
+
+        const auto checkCallback = [&checkCount](ReturnTypeCallback returnTypeCallback, const nlohmann::json & resultData)
+        {
+            if (returnTypeCallback == SELECTED && resultData.contains("count"))
+            {
+                if (resultData["count"].is_number())
+                {
+                    checkCount = resultData["count"].get<int>();
+                }
+            }
+        };
+
+        m_dBSync->selectRows(checkQuery.query(), checkCallback);
+
+        LoggingHelper::getInstance().log(LOG_DEBUG,
+                                         "Database contains " + std::to_string(policyCount) + " policies and " +
+                                         std::to_string(checkCount) + " checks");
+
+        return (policyCount > 0 || checkCount > 0);
+    }
+    catch (const std::exception& err)
+    {
+        LoggingHelper::getInstance().log(LOG_ERROR, "Error checking database contents: " + std::string(err.what()));
+        return false;
+    }
+}
+
+bool SecurityConfigurationAssessment::handleAllPoliciesRemoved()
```

Comment:
Why not just delete the databases (both sca and sca_sync) after the data clean notification like it is done in FIM?